### PR TITLE
Remove skip_if_not_multigpu decorator

### PR DESCRIPTION
**Pull request by H-Huang**
Fixes https://github.com/pytorch/pytorch/issues/54887

Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#54916 Remove skip_if_not_multigpu decorator**

`@skip_if_not_multigpu` was skipping all the tests that use it. Removed it and replaced all references with `@skip_if_lt_x_gpu(2)` which performs the same check correctly.

Differential Revision: [D27412193](https://our.internmc.facebook.com/intern/diff/D27412193)

### CODE REVIEW

File: test/quantization/test_quantize_fx.py [hunk 1]
```diff
@@ -3801,9 +3799,11 @@ def print_diffs(diffs):
         # print('----------------------')
 
     @skip_if_no_torchvision
-    @skip_if_not_multigpu
     @skipIfNoFBGEMM
     def test_resnet18_ddp(self):
+        # TODO: This test is always failings and needs to be updated
+        # Issue: https://github.com/pytorch/pytorch/issues/54979
+        raise unittest.SkipTest()
```
Comment (rohan-varma) [inline test/quantization/test_quantize_fx.py:3806]: nit: there is a decorator @unittest.skipTest that always skips the test

**Review by rohan-varma — APPROVED** [review]
Thanks for fixing!

**Comment by gcramer23** [comment]
looks good. I noticed the problem is that the func argument in def skip_if_not_multigpu. If you remove the argument it works as intended. also matches the pattern of the other defs with ```    def decorator(func):
        @wraps(func)```

File: torch/testing/_internal/common_distributed.py [hunk 2]
```diff
@@ -68,17 +68,14 @@ def wrapper(*args, **kwargs):
 
 def skip_if_not_multigpu(func):
```
Comment (gcramer23) [inline torch/testing/_internal/common_distributed.py:69]: looks good. I noticed the problem is that the func argument in def skip_if_not_multigpu. If you remove the argument it works as intended. also matches the pattern of the other defs with def decorator(func): @wraps(func)

**Review by gcramer23 — APPROVED** [review]

File: torch/testing/_internal/common_distributed.py [hunk 3]
```diff
@@ -68,17 +68,14 @@ def wrapper(*args, **kwargs):
...
-                return func(*args, **kwargs)
-            message = "Need at least {} CUDA devices".format(2)
-            TEST_SKIPS["multi-gpu"] = TestSkip(75, message)
-            sys.exit(TEST_SKIPS['multi-gpu'].exit_code)
-        return wrapper
-
-    return decorator
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        if torch.cuda.is_available() and torch.cuda.device_count() >= 2:
```
Comment (mrshenli) [inline torch/testing/_internal/common_distributed.py:73]: is this the same as `@skip_if_lt_x_gpu(2)`? If so, will it work if we consolidate all `@skip_if_not_multigpu` occurrences into  `@skip_if_lt_x_gpu(2)`?

Comment (H-Huang) [reply torch/testing/_internal/common_distributed.py:73]: Yep, great point. I was working on just removing this entirely and replacing all references to `@skip_if_not_multigpu` with `skip_if_lit_x_gpu(2)`, just need to figure out the failures for the tests that are now reenabled.

**Review by mrshenli — APPROVED** [review]
Thanks for fixing, LGTM! Let's wait for all tests. :)

**Comment by mrshenli** [comment]
One caveat is that, PR CIs won't run multi-gpu tests (4 GPUs), and that only runs on master or ci-all PRs. Is it necessary to try ci-all on this change?

**Comment by H-Huang** [comment]
Thanks for the tip @mrshenli! I will check that tests pass for https://github.com/pytorch/pytorch/pull/55084 before merging

**Comment by zhaojuanmao** [comment]
I'm wondering when those tests are disabled...?

**Comment by H-Huang** [comment]
test failure in rocm, looks like it should be skipped so adding @skip_if_rocm to `test_grad_layout_1devicemodule_1replicaperprocess`

File: torch/testing/_internal/common_distributed.py (hunk 2 above)
Comment (pmeier) [reply torch/testing/_internal/common_distributed.py]: The problem is that `skip_if_not_multigpu` used the double wrapper structure which is meant for parametrizable decorators. Removing one wrapper would have been sufficient. See my proposal in #53502.

**Pull request closed without merging**